src/services/reviewService.ts

```diff
@@ -0,0 +1,121 @@
+"use server";
+import { revalidatePath } from "next/cache";
+import { getServerSession } from "next-auth/next";
+import { authOptions } from "@/authOptions";
+import prisma from "@/db";
+
+const createReview = async (movieId: number, content: string, rating?: number | null) => {
+  const session = await getServerSession(authOptions);
+  if (!session) {
+    throw "Invalid session";
+  }
+  const newReview = await prisma.reviews.create({
+    data: {
+      userId: Number(session.user.id),
+      movieId,
+      content,
+      rating,
+    },
+  });
+
+  return newReview;
+};
+
+const deleteReviewById = async (reviewId: number | string) => {
+  const session = await getServerSession(authOptions);
+  if (!session) {
+    throw "Invalid session";
+  }
+  if (typeof reviewId === "number") {
+    const deletedReview = await prisma.reviews.delete({
+      where: {
+        id: reviewId,
+      },
+    });
+    revalidatePath(`/movie/${deletedReview.movieId}`);
+    return deletedReview;
+  } else {
+    const deletedReview = await prisma.importedReviews.delete({
+      where: {
+        id: reviewId,
+      },
+    });
+    revalidatePath(`/movie/${deletedReview.movieId}`);
+    return deletedReview;
+  }
+};
+
+const getImportedReviewsAndLocalReviewsById = async (movieId: number) => {
+  const review = await prisma.movies.findUniqueOrThrow({
+    where: {
+      id: movieId,
+    },
+    include: {
+      reviews: {
+        select: {
+          content: true,
+          id: true,
+          movieId: true,
```

Comment:
saiskohan tänne tuon user raportit kanssa sillain että 
```js 
         reports: {
            select: {
              
            },
            where: {
              creatorId: userId,
            },
```

Reply:
Ja jäin miettimään pitäiskö käyttäjän pystyä seuraamaan jostain kootusti hänen omia reporttejaan, mutta sellasta ei tähän pull requestiin kannata änkeä.